Bindings/Java/OpenSimJNI/OpenSimContext.cpp

```diff
@@ -432,4 +432,21 @@ void OpenSimContext::restoreStateFromCachedModel()
     this->setState(&(_model->updWorkingState()));
     this->realizePosition();
 }
+
+void OpenSimContext::setConnecteeName(Component& comp, AbstractSocket& socket, const std::string& newValue) {
+    // Since some socket changes can invalidate the socket altogether we'll do the change conservatively by
```

Comment:
Thanks @aseth1 I left at "some socket changes" because I can't see into the future what other scenarios may arise that invalidates (e.g. constraints), if you believe that joint change will be the only case for good I can make the change more specific. I agree that this can be more efficient but I'm trying to minimize change at this stage if possible.

Reply:
I should have caught this earlier,  but the name and signature of this method is also confusing. It would be good to give it a name that wasn't the exact name of the method on `Socket`. Perhaps it should be called something likeso`makeValidConnecteeNameChange(string newConnecteeName)` or return as `bool isValidConnecteeNameChange()`.

Which `Component` is being passed in as part of the signature? How do you ensure it is the `Component` that contains the passed in `Socket`? We could also enable you to have const access to the  `Component` that contains and owns the `Socket` using `Socket::getOwner() ` which is currently `protected`.

Would it also be better to attempt to connect the `Socket` with a connectee `Component` rather than just editing the name?

Reply:
Thanks @aseth1 👍 
Per our conversation, will make the method Socket::getOwner() public (so the owner doesn't need to be passed in) and rename it to setSocketConnecteeName()